Orange/widgets/visualize/ownomogram.py

```diff
@@ -1284,7 +1284,13 @@ def reconstruct_domain(original, preprocessed):
                 continue
             attrs[var] = None   # we only need keys
         attrs = list(attrs.keys())
-        return Domain(attrs, original.class_var, original.metas)
+
+        orig_clv = original.class_var
+        orig_data = classifier.original_data
+        values = (orig_clv.values[int(i)] for i in
+                  np.unique(orig_data.get_column_view(orig_clv)[0]))
```

Comment:
`np.unique` is $n \log n$. I think it would be better to have something like

```python
mask = np.zeros(len(orig_clv.values), dtype=bool)
column = orig_data.get_column_view(orig_clv)[0]
mask[column[np.isfinite(column)].astype(int)] = True
values = tuple(np.array(orig_clv.values)[mask])
```

(I haven't tested this code, it's just an idea.)

Reply:
I prefer the `unique`. For me, it is easier to read the code.

Reply:
I like what Janez did, but perhaps we could pack it into a function. Looking for values of discrete variables that are used in some column seems like something common.

Reply:
If @VesnaT finds the original better, we can keep it.

However, @markotoplak's idea to have a function make sense. Where to put it, how to name it? Hm, let's put it to `Orange.prepocess.remove` and name it `remove_unused_values`. Oh, no, wait, it's already there.

It is understandable that @VesnaT and I did not know about this function. Oh, no, wait, @VesnaT wrote it and I did some work on it two years ago.